feature-libs/my-account/organization/core/store/effects/b2b-user.effect.ts

```diff
@@ -46,58 +56,88 @@ export class B2BUserEffects {
 
   @Effect()
   createB2BUser$: Observable<
-    B2BUserActions.CreateB2BUserSuccess | B2BUserActions.CreateB2BUserFail
+    | B2BUserActions.CreateB2BUserSuccess
+    | B2BUserActions.CreateB2BUserFail
+    | OrganizationActions.OrganizationClearData
   > = this.actions$.pipe(
     ofType(B2BUserActions.CREATE_B2B_USER),
     map((action: B2BUserActions.CreateB2BUser) => action.payload),
     switchMap((payload) =>
       this.b2bUserConnector.create(payload.userId, payload.orgCustomer).pipe(
-        map((data) => {
+        switchMap((data) => {
           this.routingService.go({
             cxRoute: 'userDetails',
             params: { customerId: data.customerId },
           });
-          return new B2BUserActions.CreateB2BUserSuccess(data);
+          return [
+            new B2BUserActions.CreateB2BUserSuccess(data),
+            new OrganizationActions.OrganizationClearData(),
+          ];
         }),
 
         catchError((error: HttpErrorResponse) =>
-          of(
+          from([
             new B2BUserActions.CreateB2BUserFail({
               orgCustomerId: payload.orgCustomer.customerId,
               error: normalizeHttpError(error),
-            })
-          )
+            }),
+            new OrganizationActions.OrganizationClearData(),
+          ])
         )
       )
     )
   );
 
   @Effect()
   updateB2BUser$: Observable<
-    // B2BUserActions.UpdateB2BUserSuccess
-    B2BUserActions.LoadB2BUser | B2BUserActions.UpdateB2BUserFail
+    | B2BUserActions.UpdateB2BUserSuccess
+    | B2BUserActions.UpdateB2BUserFail
+    | OrganizationActions.OrganizationClearData
   > = this.actions$.pipe(
     ofType(B2BUserActions.UPDATE_B2B_USER),
     map((action: B2BUserActions.UpdateB2BUser) => action.payload),
     switchMap((payload) =>
       this.b2bUserConnector
         .update(payload.userId, payload.orgCustomerId, payload.orgCustomer)
         .pipe(
-          // TODO: Workaround for empty PATCH response:
-          // map((data) => new B2BUserActions.UpdateB2BUserSuccess(data)),
-          map(() => new B2BUserActions.LoadB2BUser(payload)),
+          // TODO: change for 'payload: data' when backend API start to return user data on PATCH
+          switchMap(() => [new B2BUserActions.UpdateB2BUserSuccess(payload)]),
           catchError((error: HttpErrorResponse) =>
-            of(
+            from([
               new B2BUserActions.UpdateB2BUserFail({
                 orgCustomerId: payload.orgCustomer.customerId,
                 error: normalizeHttpError(error),
-              })
-            )
+              }),
+              new OrganizationActions.OrganizationClearData(),
+            ])
           )
         )
     )
   );
 
+  @Effect()
+  verifyB2BUserAfterUpdate$: Observable<
```

Comment:
Verify means something generic, what about chaning it to `check` or for something more specific, like `selfEmailUpdate`?

Reply:
We could also add some description comment to it, as it's an exception worth explaining IMO.